pkgs/development/embedded/platformio/core.nix

```diff
@@ -41,6 +41,7 @@ with python3Packages; buildPythonApplication rec {
       hash = "sha256-yq+/QHCkhAkFND11MbKFiiWT3oF1cHhgWj5JkYjwuY0=";
       revert = true;
     })
+    ./disable-version-check.patch
```

Comment:
So we could just `sed -i "/check_platform_upgrade()//d" platformio/maintenance.py` and not add the patch file. WDYT?

Reply:
that was actually the solution of @adisbladis in https://github.com/NixOS/nixpkgs/pull/232356 , more specifically: `     substituteInPlace platformio/maintenance.py --replace '    check_platformio_upgrade()' ""`

Reply:
*--replace-fail and it is rock solid

Reply:
I always find substituteInPlace to be lacking for dropping entirely lines, that's why I fall back to sed in these cases. No hard opinion on the matter.

Reply:
i like `--replace-fail` (haven't heard of it before) as this will expose code changes related to the fix in upstream